repo2docker/buildpacks/conda/__init__.py

```diff
@@ -127,6 +128,50 @@ def get_build_script_files(self):
         files.update(super().get_build_script_files())
         return files
 
+    _environment_yaml = None
+
+    @property
+    def environment_yaml(self):
+        if self._environment_yaml is not None:
+            return self._environment_yaml
+
+        environment_yml = self.binder_path("environment.yml")
+        if not os.path.exists(environment_yml):
+            self._environment_yaml = {}
+            return self._environment_yaml
+
+        with open(environment_yml) as f:
+            env = YAML().load(f)
+            # check if the env file is empty, if so instantiate an empty dictionary.
+            if env is None:
+                env = {}
+            # check if the env file provided a dick-like thing not a list or other data structure.
```

Comment:
```suggestion
            # check if the env file provided a dict-like thing not a list or other data structure.
```
the schoolboy in me chuckled